pkg/clusters/types/eks/aws-operations/eks.go

```diff
@@ -0,0 +1,479 @@
+	stsClient := sts.NewFromConfig(cfg)
+	ec2Client := ec2.NewFromConfig(cfg)
+	eksClient := eks.NewFromConfig(cfg)
+	iamClient := iam.NewFromConfig(cfg)
+
+	callIdentityOutput, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return errors.Wrap(err, "failed to get caller identity")
+	}
+	createdByArn := aws.ToString(callIdentityOutput.Arn)
+
+	clusterRoleArn, nodeRoleArn, err := createRoles(ctx, iamClient, clusterName)
+	if err != nil {
+		return errors.Wrap(err, "failed to create IAM roles")
+	}
+	subnetAvZones, err := getAvailabilityZones(ctx, ec2Client, cfg.Region)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get availability zones in region %s", cfg.Region)
+	}
+
+	vpcId, subnetIDs, err := createVPC(ctx, ec2Client, subnetAvZones)
+	if err != nil {
+		return errors.Wrap(err, "failed to create VPC")
+	}
+
+	cpSgId, err := createControlPlaneSecurityGroup(ctx, ec2Client, vpcId, clusterName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create control plane security group in VPC %s", vpcId)
+	}
+
+	_, err = createCluster(ctx, eksClient, clusterName, clusterRoleArn, k8sMinorVersion, cpSgId, subnetIDs, createdByArn, tags)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create EKS cluster %s", clusterName)
+	}
+
+	activeCluster, err := waitForClusterActive(ctx, eksClient, clusterName)
+	if err != nil {
+		return errors.Wrapf(err, "failed while waiting for EKS cluster %s to become active", clusterName)
+	}
+
+	sgId, err := createNodeSecurityGroup(ctx, ec2Client, vpcId, clusterName, activeCluster.ResourcesVpcConfig.SecurityGroupIds)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create security groups")
+	}
+
+	_, kubeCfg, err := ClientForCluster(ctx, cfg, clusterName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get kube client for cluster %s", clusterName)
+	}
+
+	err = authorizeNodeGroup(kubeCfg, nodeRoleArn)
+	if err != nil {
+		return errors.Wrapf(err, "failed to authorize node group to access cluster %s", clusterName)
+	}
+
+	amiId, err := resolveAMI(ctx, ec2Client, cfg.Region, k8sMinorVersion, nodeMachineType, eksctlapi.DefaultNodeImageFamily)
+	if err != nil {
+		return errors.Wrap(err, "failed to resolve AMI")
+	}
+
+	clusterCfg := buildClusterConfig(clusterName, k8sMinorVersion, nodeMachineType, cfg.Region, amiId, subnetAvZones)
+	ng := clusterCfg.NodeGroups[0]
+	clusterCfg.VPC.ID = vpcId
+	ng.Subnets = subnetIDs
+	ng.SecurityGroups.AttachIDs = []string{sgId}
+	ng.IAM.InstanceRoleARN = nodeRoleArn
+
+	err = clusterCfg.SetClusterState(activeCluster)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create cluster state object for cluster %s", clusterName)
+	}
+
+	err = createNodeGroup(ctx, eksClient, ec2Client, clusterCfg)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create EKS node group for cluster %s", clusterName)
+	}
+
+	return nil
+}
+
+func DeleteEKSClusterAll(ctx context.Context, cfg aws.Config, clusterName string) error {
```

Comment:
Ditto here. I think we need to add comments to tell what resources are deleted here.